src/components/validation/resources/http.rs

```diff
@@ -301,32 +294,33 @@ fn spawn_input_http_client(
 #[allow(clippy::missing_const_for_fn)]
 fn spawn_output_http_server(
     config: HttpResourceConfig,
-    codec: ResourceCodec,
-    output_tx: mpsc::Sender<Vec<Event>>,
-    task_coordinator: &TaskCoordinator<Configuring>,
-    input_events: Vec<TestEvent>,
-    runner_metrics: &Arc<Mutex<RunnerMetrics>>,
+    context: HttpResourceOutputContext,
 ) -> vector_lib::Result<()> {
```

Comment:
nit: I was thinking these too could be methods in `HttpResourceOutputContext`, but whatever works at this point.

Reply:
🤦  I see that in your earlier comment now

Reply:
done in https://github.com/vectordotdev/vector/pull/20148/commits/f5ce2eb48f617ca52cc428190660c05daf70a4b2